packages/editor/src/store/actions.js

```diff
@@ -181,6 +181,15 @@ export const savePost =
 			dispatch.editPost( { content }, { undoIgnore: true } );
 		}
 
+		const preSaveOK = await applyFilters(
+			'editor.__unstablePreSavePost',
+			Promise.resolve( true ),
+			options
+		);
+		if ( ! preSaveOK ) {
+			return;
+		}
```

Comment:
Thank you for starting to implement this! 🚀 

Some further improvements would be nice:
1. The hook should be able to return/throw an error. Like `Error( 'ACF validation failed' )`. The code further down the `savePost` function will then display a notice with that error.
2. The hook should be able to inspect also the `edits`, and maybe even change them. In addition to the `options` that are already passed to it (in practice we inspect `options.isAutosave`).

The result would look like:
```js
try {
  filteredEdits = await applyFilters(
    'editor.__unstablePreSavePost',
    Promise.resolve( edits ),
    options
  );
} catch ( err ) {
  error = err;
}

if ( ! error ) {
  // do save and the SavePost hook
}
```

What do you think?

Reply:
👋 Thanks a lot for working on this, Adam! I was pointed to this PR by Jarda, as I have a somewhat related use case:

I have an entity (a `wp_navigation` postType) that is loaded via `useEntityRecords()`. The entity has a post meta field registered (which is returned by the endpoint). When saving the entity (after it has had some edits), I’d like to include the post meta, _even if the latter doesn't have any edits_. (See the bottom of [this comment](https://github.com/WordPress/gutenberg/pull/57754#issuecomment-1898988692) for more context.)

It seems that an `__unstablePreSavePost` filter -- if Jarda's suggestion of allowing modifications to `edits` is applied -- could solve that problem (although I'd need it to work in the Site Editor rather than in the Post Editor).

Reply:
> although I'd need it to work in the Site Editor rather than in the Post Editor

Do we need to add it there separately @ockham?